Detectors/TRD/reconstruction/src/EventRecord.cxx

```diff
@@ -44,192 +40,58 @@
-  sumTrackletsDigitsTriggers(trackletcount, digitcount, triggercount);
+
+  size_t digitcount = 0;
+  size_t trackletcount = 0;
   std::vector<Tracklet64> tracklets;
-  tracklets.reserve(trackletcount);
   std::vector<Digit> digits;
-  digits.reserve(digitcount);
   std::vector<TriggerRecord> triggers;
-  triggers.reserve(triggercount);
   for (auto& event : mEventRecords) {
-    //sort tracklets
-    event.sortByHCID();
-    //TODO do this sort in parallel over the events
-    tracklets.insert(std::end(tracklets), std::begin(event.getTracklets()), std::end(event.getTracklets()));
-    digits.insert(std::end(digits), std::begin(event.getDigits()), std::end(event.getDigits()));
+    event.sortTrackletsByDetector();
+    tracklets.insert(tracklets.end(), event.getTracklets().begin(), event.getTracklets().end());
+    digits.insert(digits.end(), event.getDigits().begin(), event.getDigits().end());
     triggers.emplace_back(event.getBCData(), digitcount, event.getDigits().size(), trackletcount, event.getTracklets().size());
     digitcount += event.getDigits().size();
     trackletcount += event.getTracklets().size();
   }
-  //TODO change to adopt instead of having this additional copy.
-  //
+
   pc.outputs().snapshot(o2::framework::Output{o2::header::gDataOriginTRD, "DIGITS", 0, o2::framework::Lifetime::Timeframe}, digits);
   pc.outputs().snapshot(o2::framework::Output{o2::header::gDataOriginTRD, "TRACKLETS", 0, o2::framework::Lifetime::Timeframe}, tracklets);
   pc.outputs().snapshot(o2::framework::Output{o2::header::gDataOriginTRD, "TRKTRGRD", 0, o2::framework::Lifetime::Timeframe}, triggers);
   if (generatestats) {
     accumulateStats();
     pc.outputs().snapshot(o2::framework::Output{o2::header::gDataOriginTRD, "RAWSTATS", 0, o2::framework::Lifetime::Timeframe}, mTFStats);
   }
-  mEventRecords.clear();
+
   std::chrono::duration<double, std::micro> dataReadTime = std::chrono::high_resolution_clock::now() - dataReadStart;
   LOG(debug) << "Preparing for sending and sending data took  " << std::chrono::duration_cast<std::chrono::milliseconds>(dataReadTime).count() << "ms";
 }
 
-void EventStorage::accumulateStats()
+void EventRecordContainer::accumulateStats()
 {
   int eventcount = mEventRecords.size();
   int sumtracklets = 0;
   int sumdigits = 0;
-  //std::chrono::duration sumdigittime=std::chrono::steady_clock::duration::zero();
-  //std::chrono::duration sumtracklettime=std::chrono::steady_clock::duration::zero();
   int sumdigittime = 0;
   int sumtracklettime = 0;
-  int sumtime = 0;
   uint64_t sumwordsrejected = 0;
   uint64_t sumwordsread = 0;
   for (auto event : mEventRecords) {
     sumtracklets += event.getEventStats().mTrackletsFound;
     sumdigits += event.getEventStats().mDigitsFound;
     sumtracklettime += event.getEventStats().mTimeTakenForTracklets;
     sumdigittime += event.getEventStats().mTimeTakenForDigits;
+    // OS: the two counters below are not even used anymore, are they needed?
```

Comment:
no they are not

Reply:
good, then they'll also go with the next PR